src/librbd/librbd.cc

```diff
@@ -5910,6 +5910,47 @@ extern "C" ssize_t rbd_aio_compare_and_write(rbd_image_t image, uint64_t off,
   return 0;
 }
 
+extern "C" int rbd_aio_compare_and_writev(rbd_image_t image, const struct iovec *iov,
+                                          int iovcnt, uint64_t off,
+                                          rbd_completion_t c, uint64_t *mismatch_off,
+                                          int op_flags)
+{
+  librbd::ImageCtx *ictx = (librbd::ImageCtx *)image;
+  librbd::RBD::AioCompletion *comp = (librbd::RBD::AioCompletion *)c;
+
+  auto aio_completion = get_aio_completion(comp);
+  ssize_t len = 0;
+  bufferlist bl;
+  bufferlist bl_cmp;
+  bufferlist bl_buf;
+  for (int i = 0; i < iovcnt; ++i) {
+    const struct iovec &io = iov[i];
+    len += io.iov_len;
+    if (len < 0) {
+      break;
+    }
+    
+    bl.push_back(create_write_raw(ictx, static_cast<char*>(io.iov_base), io.iov_len, aio_completion)); 
+  }
+ 
+  int r = 0;
+  if (0 >= iovcnt || len < 0 || 0 != len % 2)
```

Comment:
How is this more efficient than just using `rbd_aio_compare_and_write`? This restriction to have exactly the same number of compare iovecs as the write iovecs also seems arbitrary.